### PR TITLE
Customizable owner usage hints

### PR DESCRIPTION
**Pull request by pbetkier**
Owner sources may now define a usage hint that will be presented in the console when adding or editing a topic or a subscription.

Also refactored autocompletion as an optional override in the `OwnerSource` interface instead of a subinterface.

### CODE REVIEW

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/OwnersEndpoint.java [hunk 1]
```diff
@@ -36,9 +35,7 @@ public OwnersEndpoint(OwnerSources ownerSources) {
     @ApiOperation(value = "Lists owners from the given source matching the search string", response = List.class, httpMethod = HttpMethod.GET)
     public List<Owner> search(@PathParam("source") String source,
                               @QueryParam("search") String searchString) {
-        return ownerSources.getAutocompleteByName(source)
-                .map(s -> s.ownersMatching(searchString))
-                .orElseThrow(() -> new OwnerSourceNotFound(source));
+        return ownerSources.getAutocompletionFor(source).ownersMatching(searchString);
```
Comment (dankraw) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/OwnersEndpoint.java]: 🤔 nice sentence 👏

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/owner/CrowdOwnerSource.java [hunk 2]
```diff
@@ -50,4 +55,8 @@ public Owner get(String id) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public Optional<String> usageHint() {
+        return Optional.of("Crowd group (or groups separated by ',')");
```
Comment (pbetkier) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/owner/CrowdOwnerSource.java]: It's true. On the other hand, it's all in one place ;) Alternatively I could move these custom hints to console config json, what do you think?

Comment (adamdubiel) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/owner/CrowdOwnerSource.java]: Yes, please do.

File: hermes-console/static/js/console/owner/OwnerSelector.js [hunk 3]
```diff
@@ -42,6 +43,22 @@ angular.module('hermes.owner')
         $scope.owners = function(searchString) {
             return ownerRepository.getOwners($scope.sourceSelectModel.name, searchString);
         };
+
+        $scope.placeholder = function () {
```
Comment (dankraw) [inline hermes-console/static/js/console/owner/OwnerSelector.js]: This name may be too generic for global scope, maybe something like `ownerPlaceholder`?

Comment (dankraw) [inline hermes-console/static/js/console/owner/OwnerSelector.js]: Nevermind, I was thinking about `$rootScope`.

**Pull request merged**